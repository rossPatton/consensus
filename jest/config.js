const REGEX = {
  css: '^.+\\.css$',
  file: '^.+\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$',
  js: '^.+\\.js$',
};

const jestConfig = {
  moduleDirectories: [
    'jest',
    'node_modules',
    'src',
  ],

  // "setupFiles": [
  //   "<rootDir>/jest/__shim__/shim.js",
  //   "<rootDir>/jest/__setup__/setup.js",
  //   "<rootDir>/jest/__mocks__/global.mock.js",
  // ],

  // moduleNameMapper: {
  //   [REGEX.css]: 'identity-obj-proxy',
  //   [REGEX.file]: '<rootDir>/jest/__mocks__/file.mock.js',
  // },

  rootDir: '../',

  transform: {
    [REGEX.js]: 'babel-jest',
  },
};

module.exports = jestConfig;
