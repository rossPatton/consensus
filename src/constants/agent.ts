// import https from 'https';

// // if dev, ignore SSL. if prod, enforce cert
// // this is useful for server side fetching in development mode
// export const agent = new https.Agent({
//   rejectUnauthorized: __PROD__,
// });
