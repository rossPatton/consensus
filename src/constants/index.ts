export const categories: ts.categoryObj[] = [
  {display: 'Community', slug: 'community'},
  {display: 'Cooperative', slug: 'cooperative' },
  {display: 'Union', slug: 'union' },
  {display: 'Political', slug: 'political' },
];

export const categoryMap: ts.categoryMap = {
  'community': 'Community',
  'cooperative': 'Cooperative',
  'union': 'Union',
  'political': 'Political',
};

export const roles: ts.role[] = ['member', 'facilitator'];

export const spacesUrl = 'https://consensus.nyc3.digitaloceanspaces.com';
