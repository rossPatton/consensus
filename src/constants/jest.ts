const country = 'United States';

export const testGroup1 = {
  avatar: '',
  category: 'Union' as ts.category,
  city: 'New York',
  cityId: 1,
  country,
  countryId: 1,
  description: '',
  deletionDeadline: '',
  email: 'testgroup1@test.com',
  id: 1,
  memberName: 'member',
  modName: 'facilitator',
  name: 'Tech Workers Coalition Chapter',
  region: 'New York',
  regionId: 1,
  handle: 'tech-workers-coalition-chapter',
  sessionType: 'group' as ts.sessionType,
  type: 'private' as ts.privacyEnum,
};

export const testUser1 = {
  avatar: '',
  city: 'New York',
  cityId: 1,
  country,
  countryId: 1,
  email: 'testgroup1@test.com',
  id: 1,
  name: 'Test User',
  privateMemberships: true,
  privateRSVP: true,
  region: 'New York',
  regionId: 1,
  sessionType: 'user' as ts.sessionType,
} as ts.user;

export const testMeeting1 = {
  attendees: 1,
  host: '',
  publicRSVPS: [{}] as ts.user[],
  rsvp: { userId: 1 },
  id: 12,
  category: 'Political' as ts.category,
  groupId: 1,
  cityId: 16624,
  groupName: 'Tech Workers Coalition NYC',
  isOnline: false,
  isPrivate: false,
  description: 'Asperiores et perspiciatis fugit sint. Nihil dolore autem enim numquam dolores aliquam sed repudiandae. Eius dolorem nostrum quo aut culpa incidunt. Aut eum adipisci. Amet amet suscipit et. Voluptas animi distinctio adipisci. Nam voluptas similique quam esse magni quidem id. In aut nulla inventore facilis dicta ducimus molestias saepe. Autem voluptatibus mollitia quos ullam quidem ex consequuntur quia. Adipisci quod est incidunt quas repellendus rem voluptatem. Cumque iusto corporis velit nam nostrum commodi vel possimus velit. Quibusdam ea cumque dignissimos cumque magni voluptas ipsa natus sunt. Est qui officiis atque dolorem voluptatibus aut explicabo.',
  location: '717 Borer Oval',
  locationLink: 'http://sam.info',
  locationType: 'online',
  title: 'aut occaecati iusto nostrum ut',
  slug: 'aut-occaecati-iusto-nostrum-ut',
  date: '2020-05-02T10:47:34.393Z',
  endDate: '2020-05-02T11:01:06.513Z',
  isDraft: false,
  duration: 2,
  tag: 'Protest' as ts.meetingTypes,
  img: '',
  time: '19:00',
};

export const testMeeting2 = {
  attendees: 1,
  category: 'Cooperative' as ts.category,
  cityId: 16624,
  date: '2021-05-02T10:47:34.393Z',
  description: 'Asperiores et perspiciatis fugit sint. Nihil dolore autem enim numquam dolores aliquam sed repudiandae. Eius dolorem nostrum quo aut culpa incidunt. Aut eum adipisci. Amet amet suscipit et. Voluptas animi distinctio adipisci. Nam voluptas similique quam esse magni quidem id. In aut nulla inventore facilis dicta ducimus molestias saepe. Autem voluptatibus mollitia quos ullam quidem ex consequuntur quia. Adipisci quod est incidunt quas repellendus rem voluptatem. Cumque iusto corporis velit nam nostrum commodi vel possimus velit. Quibusdam ea cumque dignissimos cumque magni voluptas ipsa natus sunt. Est qui officiis atque dolorem voluptatibus aut explicabo.',
  duration: 2,
  endDate: '2021-05-02T12:01:06.513Z',
  groupId: 1,
  groupName: 'Test Group',
  host: '',
  id: 12,
  img: '',
  isDraft: false,
  isOnline: false,
  isPrivate: false,
  location: '717 Borer Oval',
  locationLink: 'http://sam.info',
  locationType: 'online',
  publicRSVPS: [{}] as ts.user[],
  rsvp: { userId: 1 },
  slug: 'aut-occaecati-iusto-nostrum-ut',
  tag: 'Protest' as ts.meetingTypes,
  time: '19:00',
  title: 'aut occaecati iusto nostrum ut',
};

export const testGeo = {
  city: 'New York',
  countryCode: 'us',
  handle: 'new-york',
  region: 'New York',
  regionCode: 'ny',
  postcode: 10002,
};

export const testGeoEmpty = {
  city: '',
  countryCode: 'us',
  handle: '',
  region: '',
  regionCode: '',
  postcode: 0,
};

export const testNYC = {
  id: 1,
  name: 'New York',
  country,
  countryId: 1,
  region: 'New York',
  regionId: 37,
  regionCode: 'ny',
};

export const testUserSession = {
  error: null as Error | null,
  isLoading: false,
  data: {
    isAuthenticated: false,
    profile: {
      region: 'New York',
    } as ts.user,
    qr: {},
    type: 'user',
  },
};

export const testUserSession2 = {
  error: null as Error | null,
  isLoading: false,
  data: {
    isAuthenticated: false,
    profile: {
      region: 'Los Angeles',
      privateRSVP: false,
    } as ts.user,
    qr: {},
    type: 'user',
  },
};
