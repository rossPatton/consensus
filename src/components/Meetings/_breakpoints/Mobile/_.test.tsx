import React from 'react';
import {MemoryRouter} from 'react-router';
import render from 'react-test-renderer';

import {testMeeting1, testMeeting2} from '~app/constants/jest';

import MobileMeetings from '.';

describe('component/Meetings/breakpoint/Mobile', () => {
  it('renders mobile meetings without error', () => {
    const component = render.create((
      <MemoryRouter>
        <MobileMeetings
          deleteMeeting={jest.fn()}
          meetingsToRender={[testMeeting1, testMeeting2]}
        />
      </MemoryRouter>
    ));

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders past mobile meetings', () => {
    const component = render.create((
      <MemoryRouter>
        <MobileMeetings
          deleteMeeting={jest.fn()}
          meetingsToRender={[testMeeting1, testMeeting2]}
          publishedFilter="past"
        />
      </MemoryRouter>
    ));

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders past as fallback mobile meetings', () => {
    const component = render.create((
      <MemoryRouter>
        <MobileMeetings
          deleteMeeting={jest.fn()}
          meetingsToRender={[testMeeting1, testMeeting2]}
          renderPastAsFallback
        />
      </MemoryRouter>
    ));

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
