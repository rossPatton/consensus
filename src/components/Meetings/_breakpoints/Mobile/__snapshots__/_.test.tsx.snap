// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/Meetings/breakpoint/Mobile renders mobile meetings 1`] = `
<ul
  className="text-left animated fadeInUp"
>
  <li
    className="font-semibold mb-2"
  >
    Past Meetings
  </li>
  <li
    className="flex flex-row mb-2 opacity-5"
  >
    <a
      className="max-w-6/12 min-w-6/12"
      href="/meeting/12/aut-occaecati-iusto-nostrum-ut}"
      onClick={[Function]}
    />
    <div
      className="pl-1"
    >
      <div
        className="flex mb-1 text-sm text-red-3 leading-none"
      >
        <time
          className="mr-1"
          dateTime="2020-05-02T10:47:34.393Z"
        >
          May 02 2020 | 6:47AM
        </time>
      </div>
      <h3
        className="capitalize leading-none mb-1"
      >
        <a
          className="no-underline"
          href="/meeting/12/aut-occaecati-iusto-nostrum-ut"
          onClick={[Function]}
        >
          aut occaecati iusto nostrum ut
        </a>
      </h3>
    </div>
  </li>
  <li
    className="flex flex-row mb-2 opacity-5"
  >
    <a
      className="max-w-6/12 min-w-6/12"
      href="/meeting/12/aut-occaecati-iusto-nostrum-ut}"
      onClick={[Function]}
    />
    <div
      className="pl-1"
    >
      <div
        className="flex mb-1 text-sm text-red-3 leading-none"
      >
        <time
          className="mr-1"
          dateTime="2021-05-02T10:47:34.393Z"
        >
          May 02 2021 | 6:47AM
        </time>
      </div>
      <h3
        className="capitalize leading-none mb-1"
      >
        <a
          className="no-underline"
          href="/meeting/12/aut-occaecati-iusto-nostrum-ut"
          onClick={[Function]}
        >
          aut occaecati iusto nostrum ut
        </a>
      </h3>
    </div>
  </li>
</ul>
`;
